### PR TITLE
Buildpacks: remove `securityContext.privileged` and add `dockerconfig` workspace

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubalik@gmail.com>

# Changes

- removing `securityContext.privileged : true` as it is not needed anymore 
- adding `dockerconfig` workspace, which is an optional workspace that allows providing a `.docker/config.json` file for Buildpacks lifecycle binary to access the container registry. The file should be placed at the root of the Workspace with name `config.json`.

Bumping Task version to 0.4 as this is breaking change.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: task/buildpacks/0.4/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Previous Versions
+
+For support of previous [Platform API][platform-api]s use a previous version of this task.
+
+> Be sure to also supply a compatible builder image (`BUILDER_IMAGE` input) when running the task (i.e. one that has a lifecycle that supports the platform API).
+
+| Version        | Platform API
+|----            |-----
+| [0.3](../0.3/) | [0.3][platform-api-0.3]
```
Comment (vinamra28) [inline task/buildpacks/0.4/README.md:104]: there is no version change from 0.3 to 0.4 so can you please revert this back to 0.2?
```suggestion
| [0.2](../0.2/) | [0.3][platform-api-0.3]
```

Comment (zroubalik) [reply task/buildpacks/0.4/README.md:104]: @vinamra28 isn't this the buildpacks version? if so, this PR bumps from 0.3 to 0.4

Comment (vinamra28) [reply task/buildpacks/0.4/README.md:104]: @zroubalik yeah this is buildpacks version but if you see buildpacks version 0.3 is using platform api 0.4 which will be same in 0.4 version of buildpacks, so no platform API bump whereas buildpacks version 0.2 is using platform api 0.3

Comment (zroubalik) [reply task/buildpacks/0.4/README.md:104]: ack, gotcha :)

Comment (zroubalik) [reply task/buildpacks/0.4/README.md]: fixed

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Pull request merged**